sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -694,6 +694,7 @@ object FunctionRegistry {
     expression[ArrayExcept]("array_except"),
     expression[ArrayUnion]("array_union"),
     expression[ArrayCompact]("array_compact"),
+    expression[ArrayBinarySearch]("array_binary_search"),
```

Comment:
don't register it here, it is for internal purposes only.

https://github.com/apache/spark/blob/74bea2ad18c37bc8f3721cdad733f04b301490c7/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala#L894-L915 is the right place

Reply:
Okay